### PR TITLE
ci(riot): bump to 0.17.7

### PR DESCRIPTION
**Pull request by P403n1x87**
## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed. If no release note is required, add label `changelog/no-changelog`.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

File: scripts/needs_testrun.py [hunk 1]
```diff
@@ -178,7 +178,14 @@ def get_patterns(suite: str) -> t.Set[str]:
...
         compos = suitespec["components"]
-        suite_patterns = set(suitespec["suites"].get(suite, []))
+        if suite not in suitespec["suites"]:
+            return set()
+
+        suite_patterns = set(suitespec["suites"][suite])
+
+        # Include patterns from include-always components
+        for patterns in (patterns for compo, patterns in compos.items() if compo.startswith("$")):
+            suite_patterns |= set(patterns)
```
Comment (Yun-Kim) [inline scripts/needs_testrun.py:188]: I might be missing some context - is this addition related to the riot upgrade?

**Review by Yun-Kim — COMMENTED** [review]
Just one question but lgtm otherwise

File: tests/.suitespec.json [hunk 2]
```diff
@@ -1,5 +1,9 @@
 {
     "components": {
+        "$harness": [
+            "riotfile.py",
+            "scripts/ddtest"
```
Comment (emmettbutler) [inline tests/.suitespec.json:5]: You'll want to remove these files from `core` below.

**Review by emmettbutler — APPROVED** [review]

File: scripts/needs_testrun.py (hunk 1 above)
Comment (P403n1x87) [reply scripts/needs_testrun.py:188]: Yes. Although @emmettbutler fixed this in another PR, I've added this new "syntax" to define patterns that always trigger a job essentially

**Review by Yun-Kim — APPROVED** [review]

**Pull request merged**